src/go/rpk/pkg/cli/cmd/debug/bundle_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build linux
+// +build linux
+
+package debug
+
+import (
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLimitedWriter(t *testing.T) {
+	const block = 4096
+	tests := []struct {
+		name          string
+		limit         int
+		blocksToWrite float64
+	}{{
+		name:          "it should write everything if the limit is larger than the total bytes",
+		limit:         block * 2,
+		blocksToWrite: 1.5,
+	}, {
+		name:          "it should write up to the limit if the total bytes are larger than the limit",
+		limit:         block,
+		blocksToWrite: 2,
+	}}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(st *testing.T) {
+			writer := ioutil.Discard
+			lim := &limitedWriter{
+				w:          writer,
+				limitBytes: tt.limit,
+			}
+			var writeErr error
+			remaining := tt.blocksToWrite
+			written := 0
+			for remaining > 0 {
+				mult := math.Min(remaining, 1)
+				bs := make([]byte, int(block*mult))
+				_, err := rand.Read(bs)
+				require.NoError(st, err)
```

Comment:
No need for these two lines, we aren't testing we can read, and we aren't checking the random contents.

Reply:
I like to test with random data when possible (you never know what weird corner case you'll find), but this is simple enough.